src/pudl/io_managers.py

```diff
@@ -570,10 +570,18 @@ def __init__(self, base_path: UPath, schema: pa.Schema) -> None:
 
     def dump_to_path(self, context: OutputContext, obj: pd.DataFrame, path: UPath):
         """Write dataframe to parquet file."""
-        raise NotImplementedError("This IO Manager doesn't support writing data.")
+        logger.info(f"Write df to parquet at {path}")
+        schema = Resource.from_id("hourly_emissions_epacems").to_pyarrow()
+        # TODO (daz): this writes out to a directory; if we use path.open("wb")
+        # we end up with this weird "io.UnsupportedOperation.read" error.
+        # what is it trying to read?
+        return obj.to_parquet(
+            path, engine="pyarrow", schema=schema, compression="snappy"
+        )
```

Comment:
@zschira  Maybe I don't have enough context to understand what's going on here, but I don't think `df.to_parquet()` can appending to an existing parquet files.

If we're trying to write additional data into an existing parquet file -- appending together all of the existing state-year partitions into a single file -- I think we need to use [pyarrow.parquet.ParquetWriter.write_table()](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetWriter.html#pyarrow.parquet.ParquetWriter.write_table), as we were in the monolithic parquet output before.  But instead of obtaining each incremental dataframe by doing the extract/transform for the partition, we would do it by reading in a partition's worth of data from the existing parquet files. We could/should probably avoid turning it into a pandas dataframe at all, and just read it into a pyarrow table, and then write that pyarrow table out to the big concatenated parquet file. 

```py
    else:
        ds = Datastore(**ds_kwargs)
        schema = Resource.from_id("hourly_emissions_epacems").to_pyarrow()
        epacems_path = Path(
            pudl_settings["parquet_dir"], "epacems/hourly_emissions_epacems.parquet"
        )
        if epacems_path.exists() and not clobber:
            raise SystemExit(
                "The EPA CEMS parquet file already exists, and we don't want to clobber it.\n"
                f"Move {epacems_path} aside or set clobber=True and try again."
            )

        with pq.ParquetWriter(
            where=str(epacems_path),
            schema=schema,
            compression="snappy",
            version="2.6",
        ) as pqwriter:
            for part in epacems_settings.partitions:
                year = part["year"]
                state = part["state"]
                logger.info(f"Processing EPA CEMS hourly data for {year}-{state}")
                df = pudl.extract.epacems.extract(year=year, state=state, ds=ds)
                df = pudl.transform.epacems.transform(df, pudl_engine=pudl_engine)
                pqwriter.write_table(
                    pa.Table.from_pandas(df, schema=schema, preserve_index=False)
                )
```